### DISCUSSION TITLE
Undo and Redo for Entry and TextBox

### DISCUSSION THREAD

**Discussion by nekoalin** [TomSchimansky/CustomTkinter#2010] [Q&A]
Hello, 

Is there any way to have Undo and Redo for Entry and TextBox?

**Comment by Akascape** [comment]
@nekoalin Did you try passing the `undo=True` parameter in textbox?

**Reply by nekoalin** [reply]
Hmm, look like it works for textbox, but not for entry. :<

**Reply by Akascape** [reply]
@nekoalin Yes, it will only work for textbox because it has the text widget of tkinter. You can also use `maxundo` parameter to minimise memory usage. I recommend using the textbox, but you can also implement the undo/redo in entry widget, check this: https://stackoverflow.com/questions/4146971/undo-and-redo-in-an-tkinter-entry-widget

**Reply by vihaanmody1** [reply]
redo doesnt work
